File: src/main/scala/com/example/rest/InMemoryUserRepository.java
package com.example.rest;

import java.util.Arrays;

public class InMemoryUserRepository implements UserRepository {
	private static final String[] USERNAMES = { "username", "pietro", "test" };

	@Override
	public boolean check(String username, String password) {
		return "password".equals(password) && (Arrays.binarySearch(USERNAMES, username) >= 0);
	}

}

File: src/main/scala/com/example/rest/InMemoryUserRepository.scala
package com.example.rest;

class InMemoryUserRepository extends UserRepository {
  private val USERNAMES = List("username", "pietro", "test")

  override def check(username: String, password: String) = {
    val index = USERNAMES.indexOf(username)
    "password".equals(password) && USERNAMES.contains(username)
  }
}
